client/cmdhflto.c

```diff
@@ -428,19 +482,36 @@ static int CmdHfLTODump(const char *Cmd) {
         return PM3_EINVARG;
     }
 
-// alloc memory
-// loop all blocks
-// save to file
-// free memory
+    // alloc memory
+    uint8_t *dump = calloc(dump_len, sizeof(uint8_t));
+    if (!dump) {
+        PrintAndLogEx(ERR, "error, cannot allocate memory");
+        return PM3_EMALLOC;
+    }
+
+    // loop all blocks
+    int ret_val = dumpLTO(serial_number, sizeof(serial_number), dump, true);
+    if (ret_val != PM3_SUCCESS) {
+        free(dump);
+        return ret_val; 
+    }
+
+    // save to file 
+    if (filename[0] == '\0') {
+        memcpy(filename, sprint_hex_inrow(serial_number, sizeof(serial_number)), sizeof(serial_number) * 2);
+    }
+    saveFile(filename, ".bin", dump, dump_len);
+    saveFileEML(filename, dump, dump_len, 32);
+
```

Comment:
I think you can use the fillUID function to get the uid from the first bytes.

The name convention for dump files is:   'hf_lto_xxxxxx.extension'  where xxxxx is UID (or serial number in this case.